core/iwasm/libraries/lib-wasi-threads/lib_wasi_threads_wrapper.c

```diff
@@ -31,16 +39,43 @@ typedef struct {
 static int32
 allocate_thread_id()
 {
-    static int32 thread_id = 0;
-
     int32 id;
 
     os_mutex_lock(&thread_id_lock);
-    id = thread_id++;
+    if (avail_tids.pos < 0) {
+        // Resize stack and push new thread ids
+        uint32 old_size = avail_tids.size;
+        avail_tids.size *= 2;
```

Comment:
check overflow?

Reply:
right, I'll update it to
```c
if (avail_tids.size * 2 < avail_tids.size) { // Overflow check
    os_mutex_unlock(&thread_id_lock);
    return -1;
}
avail_tids.size *= 2;
```

Reply:
Also, worth adding error message in spawn_thread so it's clear why creating a new thread couldn't be completed.